### ISSUE TITLE
UI Changed after HA April Update

### ISSUE THREAD

**Issue by aAbdman** [uvjim/linksys_velop#154]
Hello Uvjim

after Home Assistant April update the Lovelace UI for  linksys has changed
it was like this:

![Screen Shot 2022-04-06 at 10 44 13 PM](https://user-images.githubusercontent.com/94934393/162058809-d341ea5c-d652-4a33-8298-fb1995e63d95.png)
 
and After the Update it became like this:

![Screen Shot 2022-04-06 at 10 40 56 PM](https://user-images.githubusercontent.com/94934393/162058944-6f30ba59-7ca5-421a-8498-a886ac6dc0d1.png)

Do you have any suggestions?
the rest of the UI's are working fine

Thanks

**Comment by uvjim** [comment]
Yup.  I forgot to update the examples in the README.  I have fixed it in my local one... Give me a mo and I'll push the change to the README...

**Comment by uvjim** [comment]
Could you try this in card 3 please...

```yaml
type: custom:auto-entities
card:
  type: vertical-stack
card_param: cards
sort:
  method: friendly_name
  reverse: false
filter:
  include:
    - entity_id: /^binary_sensor\.velop_(?!(mesh)).*_status/
      options:
        type: custom:config-template-card
        variables:
          BUTTONS: |
            () => {
              var ret = []
              var entity_prefix = "button." + "this.entity_id".split(".")[1].split("_").slice(0, -1).join("_")
              for (var entity_id in states) {
                if (entity_id.startsWith(entity_prefix)) {
                  var entity_action = states[entity_id].attributes.friendly_name.split(':')[1].trim()
                  var entity_name = states[entity_id].attributes.friendly_name.split(':')[0].replace('Velop', '').trim()
                  ret.push({
                    'entity': entity_id,
                    'name': entity_action,
                    'tap_action': {
                        'action': 'call-service',
                        'service': 'linksys_velop.' + entity_action.toLowerCase() + '_node',
                        'service_data': {
                          'node_name': entity_name,
                        },
                        'confirmation': {
                          'text': 'Are you sure you want to reboot the ' + entity_name + ' node?'
                      }
                    }
                  })
                }
              }
              return ret
            }
          ID_CONNECTED_DEVICES: >
            "sensor." + "this.entity_id".split(".")[1].split("_").slice(0,
            -1).join("_") + "_connected_devices"
          ID_LAST_UPDATE_CHECK: >
            "sensor." + "this.entity_id".split(".")[1].split("_").slice(0,
            -1).join("_") + "_last_update_check"
          ID_MODEL: >
            "sensor." +
            "this.entity_id".split(".")[1].split("_").slice(0,-1).join("_") +
            "_model"
          ID_PARENT: >
            "sensor." +
            "this.entity_id".split(".")[1].split("_").slice(0,-1).join("_") +
            "_parent"
          ID_SERIAL: >
            "sensor." +
            "this.entity_id".split(".")[1].split("_").slice(0,-1).join("_") +
            "_serial"
          ID_UPDATE_AVAILABLE: >
            "update." +
            "this.entity_id".split(".")[1].split("_").slice(0,-1).join("_") +
            "_update"
          CONNECTED_DEVICES_TEXT: |
            (entity_id) => {
              var ret = `
            | # | Name | IP | Type |
            |:---:|:---|---:|:---:|
            `
              if (states[entity_id].attributes.devices) {
                states[entity_id].attributes.devices.forEach((device, idx) => {
                  var connection_icon
                  switch (device.type.toLowerCase()) {
                    case "wireless":
                      connection_icon = "wifi"
                      break
                    case "wired":
                      connection_icon = "ethernet"
                      break
                    case "unknown":
                      connection_icon = "help"
                      break
                  }
                  ret += "| " + (idx + 1) + " | " + device.name + ((device.guest_network) ? "&nbsp;<ha-icon icon='hass:account-multiple'></ha-icon>" : "") + " | " + device.ip + " | <ha-icon icon='hass:" + connection_icon + "'></ha-icon> |\n"
                })
              }
              return ret
            }
        entities:
          - this.entity_id
          - ${ID_CONNECTED_DEVICES}
          - ${ID_LAST_UPDATE_CHECK}
          - ${ID_MODEL}
          - ${ID_PARENT}
          - ${ID_SERIAL}
          - ${ID_UPDATE_AVAILABLE}
        card:
          type: entities
          card_mod:
            style:
              .: |
                #states { padding-top: 0px; }
              fold-entity-row:
                $:
                  template-entity-row:
                    $: |
                      state-badge { display: none; }
                      state-badge + div { margin-left: 8px !important; }
                      .info.pointer { font-weight: 500; }
                      .state { margin-right: 10px; }
          entities:
            - type: custom:button-card
              entity: this.entity_id
              size: 100%
              show_entity_picture: true
              show_last_changed: true
              show_state: true
              tap_action:
                action: none
              entity_picture: ${'/local/velop_nodes/' + states[ID_MODEL].state + '.png'}
              name: |
                [[[
                  var ret = entity.attributes.friendly_name
                  if (ret) {
                    ret = ret.replace("Velop", "").split(":")[0].trim()
                  }
                  return ret || "N/A"
                ]]]
              state_display: |
                [[[
                  return `<ha-icon 
                    icon="hass:checkbox-blank-circle"
                    style="width: 24px; height: 24px;">
                    </ha-icon>`
                ]]]
              custom_fields:
                attr_label_model: Model
                attr_model: ${states[ID_MODEL].state}
                attr_label_serial: Serial
                attr_serial: ${states[ID_SERIAL].state}
                attr_parent: >-
                  ${(states[ID_PARENT].state && states[ID_PARENT].state !=
                  'unknown') ? 'Connected to ' + states[ID_PARENT].state :
                  'N/A'}
                attr_label_ip: IP Address
                attr_ip: '[[[ return entity.attributes.ip || ''N/A'' ]]]'
                attr_update: |
                  [[[
                    var ret
                    var entity_update = 'update.' + entity.entity_id.split('.')[1].split('_').slice(0, -1).join('_') + '_update'
                    var update_available = states[entity_update].state
                    if (update_available == 'on') {
                      ret = `<ha-icon
                          icon="hass:package-up"
                          style="width: 24px; height: 24px;"
                        >
                        </ha-icon>`
                    }
                    return ret
                  ]]]
              extra_styles: >
                div[id^="attr_"] { justify-self: end; } div[id^="attr_label_"] {
                justify-self: start; margin-left: 20px; } #label, #attr_parent {
                padding-top: 25px; font-size: smaller; }
              styles:
                card:
                  - box-shadow: none
                  - padding: 16px 8px
                grid:
                  - grid-template-areas: >-
                      "n n attr_update s" "i attr_label_model attr_label_model
                      attr_model" "i attr_label_serial attr_label_serial
                      attr_serial" "i attr_label_ip attr_label_ip attr_ip" "l l
                      l attr_parent"
                  - grid-template-columns: 15% 1fr 30px 30px
                name:
                  - font-size: larger
                  - justify-self: start
                  - padding-bottom: 20px
                label:
                  - justify-self: start
                custom_fields:
                  attr_label_last_update_check:
                    - margin-left: 0px
                  attr_last_update_check:
                    - justify-self: end
                  attr_parent:
                    - justify-self: end
                  attr_update:
                    - color: darkred
                    - justify-self: end
                    - padding-bottom: 20px
                state:
                  - justify-self: end
                  - padding-bottom: 20px
                  - color: |-
                      [[[
                        return (entity.state == 'on' ? 'darkcyan' : 'darkred')
                      ]]]
            - type: custom:auto-entities
              show_empty: false
              card:
                type: custom:fold-entity-row
                head:
                  type: section
                  label: Additional Information
                padding: 0
              filter:
                include:
                  - entity_id: ${ID_LAST_UPDATE_CHECK}
                    options:
                      name: Last update check
                  - entity_id: ${ID_PARENT}
                    not:
                      state: unknown
                    options:
                      type: custom:template-entity-row
                      name: Backhaul
                      state: >-
                        {% set backhaul_info = state_attr(config.entity,
                        'backhaul') %} {% set backhaul_speed =
                        backhaul_info.speed_mbps | round(2) %} {% if
                        (backhaul_speed | string).split('.')[1] == '0' %}
                          {% set backhaul_speed = backhaul_speed | int %}
                        {% endif %} {{ backhaul_info.connection }} ({{
                        backhaul_speed }} Mbps)
            - type: custom:auto-entities
              show_empty: false
              filter:
                include:
                  - domain: button
                    entity_id: >-
                      ${"/" +
                      "this.entity_id".split(".")[1].split("_").slice(0,-1).join("_")
                      + "/"}
              card:
                type: custom:fold-entity-row
                padding: 0
                head:
                  type: section
                  label: Actions
                entities:
                  - type: buttons
                    entities: ${BUTTONS()}
            - type: section
            - type: custom:fold-entity-row
              padding: 0
              head:
                type: custom:template-entity-row
                tap_action:
                  action: fire-dom-event
                  fold_row: true
                entity: ${ID_CONNECTED_DEVICES}
                name: >-
                  {% set name = state_attr(config.entity, 'friendly_name') %} {%
                  if name %}
                    {{ name.split(':')[1].strip() }}
                  {% endif %}
              entities:
                - type: custom:hui-element
                  card_type: markdown
                  content: ${CONNECTED_DEVICES_TEXT(ID_CONNECTED_DEVICES)}
                  card_mod:
                    style:
                      .: |
                        ha-card { border-radius: 0px; box-shadow: none; }
                        ha-markdown { padding: 16px 0px 0px !important; }
                      ha-markdown$: >
                        table { width: 100%; border-collapse: collapse; }

                        tbody tr:nth-child(2n+1) { background-color:
                        var(--table-row-background-color); }

                        thead tr th, tbody tr td { padding: 4px 10px; }
```

**Comment by aAbdman** [comment]
Thank you very much 

cheers

**Comment by uvjim** [comment]
Did that work?

**Comment by aAbdman** [comment]
Yes, It did

Thank you Uvjim 🙏🏼

**Comment by uvjim** [comment]
Brilliant. I'll update the README as well then.

**Comment by uvjim** [comment]
Fixed with 462df9c
